activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -134,9 +134,9 @@ def travel_to(date_or_time)
           now = date_or_time.to_time.change(usec: 0)
         end
 
-        simple_stubs.stub_object(Time, :now, now)
-        simple_stubs.stub_object(Date, :today, now.to_date)
-        simple_stubs.stub_object(DateTime, :now, now.to_datetime)
+        simple_stubs.stub_object(Time, :now) { at(now.to_i) }
+        simple_stubs.stub_object(Date, :today) { parse(now.to_date.to_s) }
```

Comment:
> I don't think DateTime has any similar single-number in+out representation, though.

@matthewd `DateTime.jd` can take a `Rational` with the fraction part representing the time as a fraction of a day. Unfortunately there's no easy way to get that number - it may be worth adding such a method to Active Support.